lib/tlscontext.c

```diff
@@ -848,6 +854,50 @@ tls_context_setup_sigalgs(TLSContext *self)
   return TRUE;
 }
 
+#if SYSLOG_NG_HAVE_DECL_SSL_CONF_CMD
+static gboolean
+_apply_cmd_context(TLSContext *self)
+{
+  gboolean result = TRUE;
+
+  if (self->conf_cmds_list) {
+    for (GList *next_pair = self->conf_cmds_list; next_pair; next_pair = next_pair->next)
+      {
+        gchar *cmd = next_pair->data;
+        next_pair = next_pair->next;
+        gchar *value = next_pair->data;
+
+        int add_result = SSL_CONF_cmd(self->ssl_conf_ctx, cmd, value);
+        if (add_result < 1)
+          msg_error("Error setting up SSL conf context, invalid cmd or value in config list",
+                    evt_tag_str("cmd", cmd),
+                    evt_tag_str("value", value),
+                    tls_context_format_location_tag(self));
```

Comment:
IIRC you specified SSL_CONF_FLAG_SHOW_ERRORS as well. Will I receive both errors (e.g. the ones from openssl and this one too?) do we need both?

Reply:
> It's looking good. I would try to lose the conditional compilation piece and I had a few smaller nitpicks. Pretty close to being approved on my part, but I'd love to see at least some of those comments fixed (or argued against).

Sorry for the late answer, somehow I totally missed the notifications about the review comments, doing my best to answer within a day, thanks for the patience!

Reply:
SSL_CONF_FLAG_SHOW_ERRORS is not giving enough details here, unfortunately.
This message is a must-have here I guess as it explains what key-value pair is the problematic one, unfortunately, the parser cannot give proper hint about the exact location of the problem at this point, we are too deep inside the call stack and the location info is lost somewhere during the error handling flow.

The error looks like this now:

     [2022-11-22T00:42:47.034299] Error setting up SSL conf context, invalid cmd or value in config list; cmd='InvalidKey', value='ECDHE-RSA-AES128-SHA', location='/etc/syslog-ng.conf:19:3'
     [2022-11-22T00:42:47.034334] Error setting up TLS session context; tls_error='error:1414E182:SSL routines:SSL_CONF_cmd:unknown cmd name', location='/etc/syslog-ng.conf:19:3'
     [2022-11-22T00:42:47.034346] Error initializing message pipeline; plugin_name='network', location='/etc/syslog-ng.conf:15:5'

Reply:
ok, thanks.